File: Programmers/Level 2/프린터.js
function solution(priorities, location) {
    let answer = 0
    const copyPriorities = priorities.reduce((acc, cur, idx) => {
        acc.push({ [idx] : cur })
        return acc
    }, [])
    while (priorities) {
        const max = priorities.reduce((a, b) => Math.max(a, b))
        const J = copyPriorities.shift()
        const j = priorities.shift()
        if (Object.values(J)[0] === max) {
            answer += 1
            if (Object.keys(J)[0] == location) break
        } else {
            copyPriorities.push(J)
            priorities.push(j)
        }
    }
    return answer
}

File: Programmers/Level 2/프린터.py
def solution2(priorities, location):
	answer = 0
	while priorities:
		J = priorities.pop(0)
		if list(filter(lambda x: x > J, priorities)):
			priorities += [J]
			if location == 0:
				location = len(priorities) - 1
			else:
				location -= 1
		else:
			answer += 1
			if location == 0:
				return answer
			location -= 1
			continue
	return answer


def solution(priorities, location):
	answer = 0
	while priorities:
		J = priorities.pop(0)
		if (priorities and max(J, max(priorities)) == J) or (not priorities):
			answer += 1
			if location == 0:
				return answer
			location -= 1
			continue
		else:
			priorities += [J]
			if location == 0:
				location = len(priorities) - 1
			else:
				location -= 1
	return answer


def solution3(priorities, location):
	answer = 0
	while priorities:
		J = priorities.pop(0)
		flag = False
		for p in priorities:
			if p > J:
				flag = True
				break
		if flag:
			priorities += [J]
			if location == 0:
				location = len(priorities) - 1
			else:
				location -= 1
		else:
			answer += 1
			if location == 0:
				return answer
			location -= 1
			continue
	return answer


# print(solution([2, 1, 3, 2], 2))  # 1
# print(solution([1, 1, 9, 1, 1, 1], 0))  # 5
